consensus/blake3pow/consensus.go

```diff
@@ -404,8 +397,7 @@ func (blake3pow *Blake3pow) Prepare(chain consensus.ChainHeaderReader, header *t
 func (blake3pow *Blake3pow) FinalizeAtContext(chain consensus.ChainHeaderReader, header *types.Header, state *state.StateDB, txs []*types.Transaction, uncles []*types.Header, ctx int) {
 	// Accumulate any block and uncle rewards and commit the final state root
 	accumulateRewardsAtContext(chain.Config(), state, header, uncles, ctx)
-	root := state.IntermediateRoot(chain.Config().IsEIP158(header.Number(ctx)))
-	header.SetRoot(root, ctx)
+	header.SetRoot(state.IntermediateRoot(chain.Config().IsEIP158(header.Number())))
```

Comment:
(nit) This line should be squashed with prior commit.

Reply:
squashed